localstack/aws/forwarder.py

```diff
@@ -65,6 +67,36 @@ def HttpFallbackDispatcher(provider: object, forward_url_getter: Callable[[], st
     return ForwardingFallbackDispatcher(provider, get_request_forwarder_http(forward_url_getter))
 
 
+class ContextModifier(Protocol):
+    """
+    This protocol is used to apply a specific modification to a request context.
+    It does not modify the original context; it creates a newly one instead.
+    """
+
+    def modify_context(self, context: RequestContext) -> RequestContext:
+        ...
```

Comment:
i'm not sure this abstraction is really powerful or will be useful in another context other than dynamodb. i'm hesitant about introducing abstractions too early and to high-level without understanding whether there are other problems we can solve with it.